File: web/src/api/models/IntegrationConnectionResponse.ts
import type { ConnectionStatus } from './IntegrationConnection';

/**
 * Response DTO for Integration connection operations - matches backend model
 */
export interface IntegrationConnectionResponse {
  // Entity identifiers
  id?: string;
  tenantId?: string;
  version?: number;

  // Core connection fields
  name?: string;
  secretName?: string; // Updated from connectionKey
  serviceType?: string;
  fetchMode?: string;
  isDeleted?: boolean;
  lastConnectionStatus?: ConnectionStatus;
  lastConnectionMessage?: string;
  lastConnectionTest?: string; // LocalDateTime as ISO string

  // Audit fields
  createdBy?: string;
  createdDate?: string; // Instant as ISO string
  lastModifiedBy?: string;
  lastModifiedDate?: string; // Instant as ISO string

  // UI helpers
  isDuplicate?: boolean; // Flag for potential duplicates
}

File: api/integration-execution-contract/src/main/java/com/integration/execution/contract/rest/response/IntegrationConnectionResponse.java
package com.integration.execution.contract.rest.response;

import com.integration.execution.contract.model.enums.ConnectionStatus;
import com.integration.execution.contract.model.enums.ServiceType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class IntegrationConnectionResponse {
    private String id;
    private String name;
    private String secretName;
    private String tenantId;
    private Long version;

    private ServiceType serviceType;
    private Boolean isDeleted;

    private ConnectionStatus lastConnectionStatus;
    private String lastConnectionMessage;
    private Instant lastConnectionTest;

    // Audit fields
    private String createdBy;
    private Instant createdDate;
    private String lastModifiedBy;
    private Instant lastModifiedDate;
}
